extra/mariabackup/xtrabackup.cc

```diff
@@ -3892,16 +3892,18 @@ static void xb_load_single_table_tablespace(const char *dirname,
 
 	if (is_remote) {
 		RemoteDatafile* rf = new RemoteDatafile();
+		file = rf;
 		if (!rf->open_link_file(n)) {
-			die("Can't open datafile %s", name);
+			goto cant_open;
 		}
-		file = rf;
 	} else {
 		file = new Datafile();
 		file->make_filepath(".", n, IBD);
 	}
 
 	if (file->open_read_only(true) != DB_SUCCESS) {
+	cant_open:
```

Comment:
Apparently something around here is broken, because the test would fail in MemorySanitizer as well as on Microsoft Windows. I’ll debug that next week.

Reply:
I am unable to reproduce the failure with clang 19. [Luckily the test fails also on Microsoft Windows](https://buildbot.mariadb.org/#/builders/234/builds/33862/steps/12/logs/stdio), and we actually get a stack trace from the `mariadb-backup --backup` process:
```
mariabackup!ut_dbg_assertion_failed
mariabackup!backup_file_op_fail
mariabackup!recv_sys_t::parse<recv_buf,1>
mariabackup!recv_sys_t::parse_mtr<1>
mariabackup!xtrabackup_copy_logfile
mariabackup!log_copying_thread
```
The relevant part of `backup_file_op_fail()` is the following:
```c++
	if (fail) {
		ut_a(opt_no_lock);
		die("DDL operation detected in the late phase of backup."
			"Backup is inconsistent. Remove --no-lock option to fix.");
	}
```
I see that `check_if_skip_table()` invokes `check_if_fts_table()` for `FILE_DELETE`, which the `type` should be here, unless I am misreading the output. I do not see the file name, but the length would match `./test/FTS_000000000000002c_DELETED_CACHE.ibd`, for example. An obvious problem would seem to be the following:
```c++
	case FILE_DELETE:
		fail = !check_if_skip_table(
                                filename_to_spacename(name, len).c_str())
                       && !check_if_fts_table(reinterpret_cast<const char*>(name));
                msg("DDL tracking : delete %u \"%.*s\"", space_id, int(len), name);
		break;
```
The function `check_if_fts_table()` expects a NUL terminated string. Here, the string should be followed by the end-of-mini-transaction indicator `\x00` or `\x01`, depending on the value of the sequence bit. 63a7e4c96b6ef0a3a37a711328644056c32bb68d should address this.